modules/ding_nodelist/ding_nodelist.admin.inc

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * @file
+ * Administration interface for nodelist configuration.
+ */
+
+/**
+ * General settings form.
+ */
+function ding_nodelist_settings_form() {
+  $options = node_type_get_names();
+
+  if (empty($options)) {
+    drupal_set_message(t('No content types were found, please add one first.'), 'error', FALSE);
+  }
+
+  $form['ding_nodelist_node_types'] = array(
+    '#type' => 'checkboxes',
+    '#title' => t('Available content types'),
+    '#options' => $options,
+    '#default_value' => variable_get('ding_nodelist_node_types', array()),
+    '#description' => t('Choose content types avaialable for display'),
+  );
+
+  // Options as defined by panels/plugins/cache/simple.inc
+  $options = drupal_map_assoc(
+    array(
+      15, 30, 60, 120, 180, 240, 300, 600, 900, 1200, 1800, 3600,
+      7200, 14400, 28800, 43200, 86400, 172800, 259200, 345600, 604800),
+    'format_interval');
+  $form['ding_nodelist_cache_default_time'] = array(
+    '#type' => 'select',
+    '#options' => $options,
+    '#title' => t('Default cache life time'),
+    '#default_value' => variable_get('ding_nodelist_cache_default_time', NL_CACHE_LIFETIME),
+    '#description' => t('Lists will be cached by default for this period of time (seconds)'),
+  );
+
+  $form['ding_nodelist_autoscroll_delay'] = array(
+    '#type' => 'fieldset',
+    '#collapsible' => FALSE,
+    '#title' => t('Autoscroll delay'),
+    '#description' => t('The delay to show an item for various nodelist widgets (in milliseconds).'),
+  );
+
+  $autoscrolling_widgets = array(
+    NL_WIDGET_CAROUSEL => array('Carousel', 3000),
+    NL_WIDGET_HOR_ACCORDION => array('Horizontal accordion', 3000),
```

Comment:
Please add keys to each configuration. That makes it easier for people to know what each value represents e.g.

NL_WIDGET_CAROUSEL => array('title' => 'Carousel', 'autoscroll_delay' => 3000),